crates/compiler/builtins/roc/Hash.roc

```diff
@@ -0,0 +1,68 @@
+interface Hash
+    exposes [
+        Hash,
+        Hasher,
+        hash,
+        addBytes,
+        addU8,
+        addU16,
+        addU32,
+        addU64,
+        addU128,
+        addI8,
+        addI16,
+        addI32,
+        addI64,
+        addI128,
+        complete,
+    ] imports []
+
+## A value that can hashed.
+Hash has
+    ## Hashes a value into a [Hasher].
+    ## Note that [hash] does not produce a hash value itself; the hasher must be
+    ## [complete]d in order to extract the hash value.
+    hash : a, hasher -> hasher | a has Hash, hasher has Hasher
```

Comment:
Actually, this should be flipped to `hash : hasher, a -> hasher` to match pipeline ordering and the `Hasher` members. Otherwise it's harder to use immediates.

Reply:
Will fix in a follow up to avoid eating CI